sanity-io/structured-content-2022
# Frontpage header

# Frontpage header

## Intent

Implement the page-specific header of the frontpage, from the first iteration of the design.

## Description

Splits the first SectionBlock of the front page out as a separate component, and implements HTML and CSS for that component based on the current design sketches in Figma. Behavior for narrower screens is simplified, no finished sketches for that quite yet.

Although the name "ConferenceHeader" has "Header" in it, I don't expect it to be common across all pages. Thus I chose to move it out of the `<header>` element, which corresponds to the `banner` ARIA role.

Not sure about the placement of this component in the file structure. I see there are some things called "pageResources". Are they intended to be components that are currently only used on one page, and then if you want to reuse the component somewhere else you have to move it into `web/components`? Or is it something else?

I'm thinking that if the front page is changed into a "landing page" in Knut's regime, then the "page-specific" concept might no longer apply. This element may then end up corresponding to some "Section" instead… but this still seems unclear.

I'm trying to focus on the frontend parts here, and not so much on the relation between the actual underlying content and the structure in Sanity. Specifically, the start/end date contents do not correspond to the Figma sketch; I left the old format helper function intact. See e.g. https://github.com/sanity-io/structured-content-2022/issues/7

## Testing

1. `docker-compose up --build web`
2. Open http://localhost:3000/home
3. Verify that the section between "Home" etc and "Sign up for Structured Content 2022" looks OK
